### DISCUSSION TITLE
How to choose the proper record selector

### DISCUSSION THREAD

**Discussion by suhl79** [airbytehq/airbyte#32748] [Connector Questions]
I'm struggling to figure out the proper record selector for the following JSON response when using Connector Builder. 
Record selector set to getUris results with "No records could be extracted from responses"
Record selector set to response, getUris results with "ERROR string indices must be integers"
Record selector set to response, * results with "ERROR list indices must be integers or slices, not str"
I already tried all the combinations of record selectors I could imagine. What am I missing to select these string values from this array and get them recognized as proper records by Airbyte? This seems to be trivial but not in this case.

Response:

```javascript
[
  {
    "response": {
      "getUris": [
        "/product/1000-1199",
        "/product/1200-1399",
        "/product/1400-1599",
        "/product/1600-1799",
        "/product/1800-1999",
        "/product/2000-2199",
        "/product/2200-2399",
        "/product/2400-2599",
        "/product/2600-2799",
        "/product/2800-2999",
        "/product/3000-3199",
        "/product/3200-3399",
        "/product/3400-3599",
        "/product/3600-3799",
        "/product/3800-3999",
        "/product/4000-4199",
        "/product/4200-4399",
        "/product/4400-4599",
        "/product/4600-4799",
        "/product/4800-4999",
        "/product/5000-5199",
        "/product/5200-5399",
        "/product/5400-5599",
        "/product/5600-5799",
        "/product/5800-5999",
        "/product/6000-6199",
        "/product/6200-6399",
        "/product/6400-6599",
        "/product/6600-6612,6614-6800",
        "/product/6801-6928,6930-7001",
        "/product/7002-7059,7066-7207",
        "/product/7208-7319"
      ]
    }
  }
]
```

**Comment by marcosmarxm** [comment]
@suhl79 it looks the response is the path to the current object data, correct?
Is it possible to share the API documentation and the connector YAML file?

**Comment by dossy** [comment]
I'm running into this same issue, where the response has an array of strings and not an array of objects, and I cannot figure out the dpath expression to use with Airbyte that will let me select it as my record selector.

I was expecting to use this stream's response to fetch an array of IDs which I would use as a parent stream for another stream that will fetch the details for each of these IDs.

Is this currently possible in Airbyte using the low-code connector builder?  A recipe/tutorial on how to do this would be really helpful.

**Comment by andreimcriss** [comment]
I have a similar issue. I need to read "meter.serials" from my API and the response looks like this:

`{
  "status": 200,
  "body": {
    "routes": [
      {
        "code": "000001",
        "name": "TestName",
        "meter.serials": [
          "18ML914414",
          "19BF087224",
          "19BA490899",
          "19BF086977",
          "20ML938543",
          "19BA490900"]
       }
     ]
     }
}
`

[a comment by an automated account is not shown]
